### ISSUE TITLE
kube-proxy NodePort externalTrafficPolicy=Local is broken

### ISSUE THREAD

**Issue by brb** [cilium/cilium#9284]
@olemarkus has reported that the kube-proxy NodePort's `externalTrafficPolicy=Local` is broken on `v1.6`. By broken, we mean that the source IP addr gets SNAT'ed by the `cilium_host` IP addr instead of the original source IP addr being preserved.

The fix for `v1.5.7` - https://github.com/cilium/cilium/pull/8944 (not backported to `v1.6`).

**Comment by brb** [comment]
The following rule SNATs a request:
```
-A CILIUM_POST_nat ! -s 10.0.0.0/8 -o cilium_host -m conntrack --ctstate DNAT -m comment --comment "cilium hostport cluster masquerade" -j SNAT --to-source 10.217.1.106
```

**Comment by brb** [comment]
The issue was fixed by #9561. Will close the issue once I have added the integration tests.
